salt/modules/heat.py

```diff
@@ -0,0 +1,870 @@
+            return ret
+        except Exception as ex:  # pylint: disable=W0703
+            log.error("Delete failed {0}".format(ex))
+            ret['result'] = False
+            ret['comment'] = '{0}'.format(ex)
+            return ret
+
+        if stack_status == 'DELETE_FAILED':
+            ret['result'] = False
+            ret['comment'] = 'Deleted stack FAILED\'{0}\'{1}.'.format(name, msg)
+        else:
+            ret['comment'] = 'Deleted stack {0}.'.format(name)
+    return ret
+
+
+def create_stack(name=None, template_file=None, enviroment=None,
+                 parameters=None, poll=0, rollback=False, timeout=3600,
+                 profile=None):
+    '''
+    Create a stack (heat stack-create)
+
+    name
+        Name of the new stack
+
+    template_file
+        File of template
+
+    enviroment
+        File of enviroment
+
+    parameters
+        Parameter dict used to create the stack
+
+    poll
+        Poll and report events until stack complete
+
+    rollback
+        Enable rollback on create failure
+
+    timeout
+        Stack creation timeout in minutes
+
+    profile
+        Profile to build on
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' heat.create_stack name=mystack \\
+                 template_file=salt://template.yaml \\
+                 enviroment=salt://enviroment.yaml \\
+                 parameters="{"image": "Debian 8", "flavor": "m1.small"}" \\
+                 poll=5 rollback=False timeout=60 profile=openstack1
+
+    '''
+    h_client = _auth(profile)
+    ret = {
+        'result': True,
+        'comment': ''
+    }
+    if not parameters:
+        parameters = {}
+    if template_file:
+        template_tmp_file = salt.utils.mkstemp()
+        tsfn, source_sum, comment_ = __salt__['file.get_managed'](
+            name=template_tmp_file,
+            template=None,
+            source=template_file,
+            source_hash=None,
+            user=None,
+            group=None,
+            mode=None,
+            saltenv='base',
+            context=None,
+            defaults=None,
+            skip_verify=False,
+            kwargs=None)
+
+        template_manage_result = __salt__['file.manage_file'](
+            name=template_tmp_file,
+            sfn=tsfn,
+            ret=None,
+            source=template_file,
+            source_sum=source_sum,
+            user=None,
+            group=None,
+            mode=None,
+            saltenv='base',
+            backup=None,
+            makedirs=True,
+            template=None,
+            show_changes=False,
+            contents=None,
+            dir_mode=None)
+        if template_manage_result['result']:
+            with salt.utils.fopen(template_tmp_file, 'r') as tfp_:
+                tpl = tfp_.read()
+                salt.utils.safe_rm(template_tmp_file)
+                try:
+                    if isinstance(tpl, six.binary_type):
+                        tpl = tpl.decode('utf-8')
```

Comment:
Is `utf-8` always the proper encoding type? Is it concievable that we'll get non-utf-8 responses back here?